File: examples/Basic/src/lib/geocore.ts
export type GeoPoint = {
  id: string;
  unicode: string;
  lat: number;
  lng: number;
  geom: string;
};

export const allArequipa = async (totalPoints: number): Promise<GeoPoint[]> => {
  const res = await fetch(
    `https://geocore.innovalab.minsky.cc/api/v1/group/ee73a646-0066-4f4a-8ee9-358e77ebba7f?limit=${totalPoints}`,
  );

  const data = await res.json();

  return data as GeoPoint[];
};

File: lib/geocore.dart
class GeoPoint {
  String? id;
  String? unicode;
  double? lat;
  double? lng;
  String? geom;

  GeoPoint({
    this.id,
    this.unicode,
    this.lat,
    this.lng,
    this.geom,
  });

  GeoPoint.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    unicode = json['unicode'];
    lat = json['lat'];
    lng = json['lng'];
    geom = json['geom'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['unicode'] = this.unicode;
    data['lat'] = this.lat;
    data['lng'] = this.lng;
    data['geom'] = this.geom;
    return data;
  }
}
